tests/test_h5dict.py

```diff
@@ -78,18 +81,114 @@ def work() -> None:
     await asyncio.sleep(0.5)
 
 
+@pytest.mark.asyncio
+@pytest.mark.allow_errors_in_log
+async def test_lock() -> None:
+    app = FastAPI()
+
+    app.include_router(router, prefix="/results")
+
+    rw_lock = RWLockFair()
+    r_lock = rw_lock.gen_rlock()
+    r_lock2 = rw_lock.gen_rlock()
+    w_lock = rw_lock.gen_wlock()
+
+    def get_data() -> Tuple[dict[str, Any], ContextManager[Any]]:
+        dt = np.dtype({"names": ["a", "b"], "formats": [float, int]})
+        arr = np.array([(0.5, 1)], dtype=dt)
+        data = {
+            "live": 34,
+            "other": {"third": [1, 2, 3]},  # only _attrs allowed in root
+            "other_attrs": {"NX_class": "NXother"},
+            "spaced group": {"space ds": 4, "space ds_attrs": {"bla": 5}},
+            "spaced group_attrs": {"spaced attr": 3},
+            "image": np.ones((1000, 1000)),
+            "specialtyp": np.ones((10, 10), dtype=">u8"),
+            "specialtyp_attrs": {"NXdata": "NXspecial"},
+            "composite": arr,
+            "composite_attrs": {"axes": ["I", "q"]},
+            "hello": "World",
+            "_attrs": {"NX_class": "NXentry"},
+        }
+        return data, r_lock  # type: ignore
+
+    app.state.get_data = get_data
+
+    config = uvicorn.Config(app, port=5000, log_level="debug")
+    server = uvicorn.Server(config)
+    server_task = asyncio.create_task(server.serve())
+    while server.started is False:
+        await asyncio.sleep(0.1)
+
+    def work() -> None:
+        f = h5pyd.File(
+            "/", "r", endpoint="http://localhost:5000/results", timeout=1, retries=0
+        )
+        w_lock.acquire()
+        logging.info("Write Lock acquired")
+        with pytest.raises(KeyError):
+            f["live"]
+        with pytest.raises(KeyError):
+            f["other"]
+        with pytest.raises(KeyError):
+            f["spaced group"]
+        with pytest.raises(KeyError):
+            f["spaced group/space ds"]
+        with pytest.raises(KeyError):
+            f["image"]
+        with pytest.raises(KeyError):
+            f["specialtyp"]
+        with pytest.raises(KeyError):
+            f["hello"]
+        with pytest.raises(KeyError):
```

Comment:
Do we gain much from testing that all keys are unreachable? Maybe a dataset and a group is sufficient?